components/atom/imagePanel/src/index.js

```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+
+const BASE_16 = 16
+const HORIZONTAL_ALIGNMENTS = {
+  LEFT: 'left',
+  CENTER: 'center',
+  RIGHT: 'right'
+}
+
+const VERTICAL_ALIGNMENTS = {
+  TOP: 'top',
+  CENTER: 'center',
+  BOTTOM: 'bottom'
+}
+
+const TYPES = {
+  RESIZED: 'resized'
+}
+
+const hexToRgb = function (hex) {
+  const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i
+  hex = hex.replace(shorthandRegex, (m, r, g, b) => r + r + g + g + b + b)
+
+  var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
+  return result ? {
+    r: parseInt(result[1], BASE_16),
+    g: parseInt(result[2], BASE_16),
+    b: parseInt(result[3], BASE_16)
+  } : {}
+}
+
+const getClassNames = function ({verticalAlign, horizontalAlign, classNames, resized}) {
+  return classnames(
+    'sui-AtomImagePanel',
+    `sui-AtomImagePanel--v-${verticalAlign}`,
+    `sui-AtomImagePanel--h-${horizontalAlign}`,
+    resized && `sui-AtomImagePanel-resized`,
+    classNames
+  )
+}
+
+const getStyles = function ({placeholderColor, overlayColor, src, overlayAlpha}) {
+  const url = `url(${src})`
+  if (overlayColor) {
+    const overlayColorRgb = hexToRgb(overlayColor)
+    const rgba = `rgba(${overlayColorRgb.r}, ${overlayColorRgb.g}, ${overlayColorRgb.b}, ${overlayAlpha})`
+    var gradient = `linear-gradient(${rgba}, ${rgba})`
+  }
+  return {
+    backgroundImage: gradient ? `${gradient}, ${url}` : url,
+    backgroundColor: placeholderColor
+  }
+}
+
+const AtomImagePanel = function ({src, children, ...props}) {
+  return (
+    <div className={getClassNames(props)} style={getStyles({src, ...props})}>
+      {children}
+    </div>
+  )
+}
+
+AtomImagePanel.displayName = 'AtomImagePanel'
+
+AtomImagePanel.propTypes = {
+  className: PropTypes.string,
+  children: PropTypes.node.isRequired,
+  /**
+   * Background image
+   */
+  src: PropTypes.string.isRequired,
+  /**
+   * Background color
+   */
+  placeholderColor: PropTypes.string.isRequired,
+  /**
+   * Gradient color
+   */
+  overlayColor: PropTypes.string,
+  /**
+   * Gradient opacity
+   * @type float between 0 and 1
+   */
+  overlayAlpha: PropTypes.number,
+  /**
+   * Background position x
+   */
+  horizontalAlign: PropTypes.oneOf(Object.values(HORIZONTAL_ALIGNMENTS)),
+  /**
```

Comment:
The text is always black. 
What happened we don't provide the overlayColor  and it has it's default #000. It's black text on dark background. Have tried ?

```js
<AtomImagePanel src={imageSrcLong} overlayAlpha={0.9} />
```

Reply:
we'll create a function in sass to fix this

Reply:
Shouldn't the child component be able to render itself? I mean, we do not just set an image as background for an label, there can be any kind of component inside this container, which means that it could happen that if the component itself has a white background, but the overlay is dark, that the user sees nothing as we would set the font color to white.

In your example, I would add an class to the component that sets the color to white as we already know that the background is dark